### PR TITLE
updated component and style guide

### PR DESCRIPTION
**Pull request by jpvajda**
## Description
- updated the component guide to reference to the announcement banner documentation
- added a TOC to the component guide
- updated the style guide to include how to updated TOCs in markdown files
- add split instructions to contributing file

### CODE REVIEW

**Comment by jpvajda** [comment]
I'm going to add how to run a split test to this PR.. so don't merge just yet.

**Comment by jpvajda** [comment]
We need to discuss where split.io documentation should go before we merge this in.

**Review by zstix — APPROVED** [review]

**Pull request merged**
